js/piwik.js

```diff
@@ -4127,7 +4129,7 @@ if (typeof window.Piwik !== 'object') {
 
                 for (index = 0; index < configCookiesToDelete.length; index++) {
                     cookieName = getCookieName(configCookiesToDelete[index]);
-                    if (0 !== getCookie(cookieName)) {
+                    if (excluding.indexOf(cookieName) !== -1 && 0 !== getCookie(cookieName)) {
```

Comment:
I think here we should always exclude `consent_removed` and only delete it when it was specifically requested using `deleteCookie('consent_removed')` maybe?

Reply:
same for `consent` cookie probably? Or actually not sure if we should always keep them... hard to say as when calling `disableCookies()` it would still keep it. On the other site we need to keep track of whether consent was given or removed even when cookies are disabled... I reckon we can leave it for now like you implemented it.

Reply:
I think it should always keep them, since you can call `this.deleteCookies()` by itself too, and if a matomo user or plugin did that, it would remove the visitor's opt-out/consent status.